src/main/java/org/jspecify/nullness/NonNull.java

```diff
@@ -23,16 +23,75 @@
 import java.lang.annotation.Target;
 
 /**
- * Indicates that the annotated type usage is considered to exclude {@code null} as a value (used
- * infrequently, as we typically apply {@link NullMarked @NullMarked} to some enclosing element
- * instead).
+ * Indicates that the annotated type usage is considered to exclude {@code null} as a value; rarely
+ * needed within {@linkplain NullMarked null-marked} code.
+ *
+ * <p>This annotation has two primary purposes:
+ *
+ * <ul>
+ *   <li>To mark any sporadic non-null type usages inside a scope that is not ready to be fully
+ *       {@link NullMarked} yet.
+ *   <li>To perform a <i>non-null projection</i> of a type variable, explained below.
+ * </ul>
  *
  * <p>For a guided introduction to JSpecify nullness annotations, please see the <a
- * href="https://jspecify.dev/user-guide.html">user guide</a>.
+ * href="http://jspecify.org/docs/user-guide">User Guide</a>.
```

Comment:
Because users will at times need to remember `import org.jspecify` (which I am sure is the right choice), I want to minimize the appearances of the `.dev` address as much as possible. I don't remember why it exists. There's an issue filed about fixing that redirect direction.